foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/internal/databaseaccess/DatabasePlatform.java

```diff
@@ -1102,7 +1102,7 @@ public int getJDBCType(DatabaseField field) {
      */
     public int getJDBCType(Class<?> javaType) {
         if (javaType == null) {
```

Comment:
By bug description it looks like PostgreSQL specific issue. Other DBs like MySQL, Oracle allows store UUID type into VARCHAR columns. I don't think, that this is right place for this change. I'd like recommend `org.eclipse.persistence.platform.database.PostgreSQLPlatform#getJDBCType`.